### ISSUE TITLE
[Reanimated] Babel plugin exception: TypeError: (0 , types_12.cloneNode) is not a function

### ISSUE THREAD

**Issue by ChristopherGabba** [software-mansion/react-native-reanimated#6006]
### Description

I am trying to compile my expo-51 (managed workflow) app and I am getting the error
 
" node_modules/react-native-gesture-handler/src/handlers/gestures/hoverGesture.ts: /Users/christophergabba/Documents/ReelFeelApp/node_modules/react-native-gesture-handler/src/handlers/gestures/hoverGesture.ts: [Reanimated] Babel plugin exception: TypeError: (0 , types_12.cloneNode) is not a function"

### Steps to reproduce

1. Use the[ react-native-ignite ](https://github.com/infinitered/ignite) boilerplate
2. Choose NPM instead of yarn
3. Upgrade the project to Expo-51
4. Compile a Development Build for iOS

I solved the problem by installing all the dependencies with yarn. As soon as I use npm to install everything, it throws this error.

### Snack or a link to a repository

Working on it now

### Reanimated version

3.10.1

### React Native version

0.74.1

### Platforms

iOS

### JavaScript runtime

Hermes

### Workflow

Expo Dev Client

### Architecture

Fabric (New Architecture)

### Build type

Debug app & dev bundle

### Device

Real device

### Device model

iPhone 12

### Acknowledgements

Yes

**Comment by AdityaAA2004** [comment]
I am also encountering the same issue when running the dev client on iOS. I have the same specs as mentioned above, and react native version 0.74.1

**Comment by ChristopherGabba** [comment]
@AdityaAA2004 It works when I run it `yarn` and have a `yarn.lock` file, but not when I use `npm`.

**Comment by AdityaAA2004** [comment]
Ok, so, does the App build on the expo dev client without this error when we use the yarn package manager?

**Comment by ChristopherGabba** [comment]
@AdityaAA2004 Yes it does for me. I cannot use NPM in my project until this issue is resolved.

**Comment by ldurrenberger** [comment]
any update on this? running into the same issue.

**Comment by ldurrenberger** [comment]
@AdityaAA2004

**Comment by AdityaAA2004** [comment]
I just try restarting the app, or delete it from the simulator and re-install all dependencies. Or, I re-create the development build again. I am not able to get a permanent fix.

**Comment by fruitybits** [comment]
using yarn solved a similar issue for me -> `node_modules/react-native-gesture-handler/src/handlers/gestures/GestureDetector.tsx: (0 , types_1.cloneNode) is not a function`

**Comment by almeidapaulooliveira** [comment]
Having the same problem building Android Expo

**Comment by robdubparker** [comment]
Same issue here building `web`: 

`SyntaxError: components/ParallaxScrollView.tsx: /[path to my project]/components/ParallaxScrollView.tsx: [Reanimated] Babel plugin exception: TypeError: (0 , types_12.cloneNode) is not a function`

**Comment by sunboykenneth** [comment]
Any solution? Also encounter this issue.

**Comment by pridapablo** [comment]
Same here!

**Comment by robdubparker** [comment]
Ok, this issue was occurring for me due to having circular dependencies in my project code, causing undefined values where values should exist.  I used the `madge` package to detect existing circular dependencies and manually removed them.  After these were removed/resolved, the error no longer occurred.

Hope this helps.

**Comment by shlomiro** [comment]
same here!

**Comment by tjzel** [comment]
Package management is difficult in React Native and `npm` tends to have the most issues with it in general. We highly encourage you to use `yarn` which the most of React Native community uses with great success.

Unfortunately, we don't have enough manpower to troubleshoot issues across different package managers. Could you explain to me why is using `yarn` not an option for you?

**Comment by ChristopherGabba** [comment]
@tjzel Well I am using yarn to solve this problem in the interim. The two big package managers at the moment are `npm` and `yarn` and I've never ran across a package where only one would work. Granted, it's not holding up my project, but do feel like this should be resolved. For someone who starts their whole app with `npm` and uses it for a while then adds this package and it doesn't work, it's a pain in the butt to troubleshoot and then ultimately cut ties with `npm` and swap to `yarn`.

**Comment by tjzel** [comment]
The fact is it that doesn't happen to all the `npm` users. Most of the issues of this kind I had been troubleshooting were caused by `npm` incorrectly pulling packages from a global cache.

@ChristopherGabba if the issue is still happening to you, could you see if on `npm cache clean` works for you when you're not using `yarn`?

I guess it wouldn't be bad to at least suggest some steps in the exception thrown in the Reanimated Babel plugin.

**Comment by ChristopherGabba** [comment]
@tjzel Sure thing, I'll try to here shortly.

**Comment by mohanataraj** [comment]
> The fact is it that doesn't happen to all the `npm` users. Most of the issues of this kind I had been troubleshooting were caused by `npm` incorrectly pulling packages from a global cache.
> 
> @ChristopherGabba if the issue is still happening to you, could you see if on `npm cache clean` works for you when you're not using `yarn`?
> 
> I guess it wouldn't be bad to at least suggest some steps in the exception thrown in the Reanimated Babel plugin.

happened to me... same issue

**Comment by ThomasHoussin** [comment]
I had the same problem. `npm cache clean` didn't help (I tried `npm cache verify` before and did `npm cache clean --force`). 

Switching to yarn solved the issue.

**Comment by Yossitrx** [comment]
Switching to yarn solved the issue.

**Comment by codekojo** [comment]
Having the same issue

**Comment by tjzel** [comment]
I suggest following the steps to fis this from this issue https://github.com/software-mansion/react-native-reanimated/issues/5883
